searx/webapp.py

```diff
@@ -1133,23 +1133,26 @@ def __init__(self, app):
 
             base_url = urlparse(settings['server']['base_url'])
             self.script_name = base_url.path
+            if self.script_name.endswith('/'):
+                # remove trailing slash to avoid infinite redirect on the index
+                # see https://github.com/searx/searx/issues/2729
+                self.script_name = self.script_name[:-1]
             self.scheme = base_url.scheme
             self.server = base_url.netloc
 
     def __call__(self, environ, start_response):
-
-        script_name = self.script_name or environ.get('HTTP_X_SCRIPT_NAME', '')
+        script_name = self.script_name or environ.get('HTTP_X_SCRIPT_NAME', None)
         if script_name:
             environ['SCRIPT_NAME'] = script_name
```

Comment:
Without this PR, `environ['SCRIPT_NAME']` is set to `/` instead of an empty string.

Without uwsgi, the value is `None`.

Reply:
Without testing: `if self.script_name.endswith('/')` makes sense to me. Just for my knowledge: is the None in `environ.get('HTTP_X_SCRIPT_NAME', None)` needed to fix the issue ... BTW you can write `environ.get('HTTP_X_SCRIPT_NAME')`

Reply:
Done.

Reply:
Thanks .. but my questions remind; it is only the `if self.script_name.endswith('/')` what fixes the issue .. right?

Reply:
( I have removed this change. )